File: LeetCode/Min Cost Climbing Stairs.cpp
class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        vector<int> mincost(cost.size()+1, 0);
        mincost[0] = cost[0];
        mincost[1] = min(cost[0]+cost[1], cost[1]);
        
        cost.push_back(0);
        
        for(int i=2; i < cost.size(); i++) {
            mincost[i] = min(mincost[i-1] + cost[i], mincost[i-2] + cost[i]);
        }
        
        return mincost[cost.size()-1];
    }
};

File: LeetCode/Min Cost Climbing Stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        minCostStair = []
        
        minCostStair.append(cost[0])
        minCostStair.append(cost[1])
        
        for i in range(2, len(cost)):
            minCostStair.append(cost[i] + min(minCostStair[i-1], minCostStair[i-2]))
        
        return min(minCostStair[len(cost)-1], minCostStair[len(cost)-2])